project/src/main/webapp/scripts/createNewAssignment.js

```diff
@@ -0,0 +1,70 @@
+async function createNewAssignment() {
+    const title = document.getElementById("title").value;
+    const description = document.getElementById("description").value;
+    const maxPoints = document.getElementById("maxPoints").value;
+    const dueDate = document.getElementById("dueDate").value;
+    const dueTime = document.getElementById("dueTime").value;
+
+    const yearMonthDayHoursMinutes = getYearMonthDayHoursMinutes(dueDate, dueTime);
```

Comment:
I think you can simplify a lot of this if you construct a `Date` directly, as the time and date inputs should give you properly formatted dates and times:
```
const dueDateTime = new Date(`${document.getElementById("dueDate").value} ${document.getElementById("dueTime").value}`);
const year = dueDateTime.getFullYear();
const month = dueDateTime.getMonth();
...
```

Reply:
Yeah, I think this could be simpler in some ways but when I looked into it almost all the guides said to not directly create the Date object from a string because of differences in browsers and also because Google Classroom needs dates in UTC time I have to convert the local date into a UTC date.